src/stdinreader.cpp

```diff
@@ -467,6 +467,24 @@ int cmdOutputThreadFunc(void *)
 	return 0;
 }
 
+static bool checkPlayerIdentityMatchesString(const EcKey& identity, const std::string& playerIdentityStrCopy)
+{
+	if (identity.empty())
+	{
+		return (playerIdentityStrCopy == "0"); // special case for empty identity, in case that happens...
+	}
+
+	// Check playerIdentityStrCopy versus both the (b64) public key and the public hash
+	std::string checkIdentityHash = identity.publicHashString();
+	std::string checkPublicKeyB64 = base64Encode(identity.toBytes(EcKey::Public));
+	if (playerIdentityStrCopy == checkPublicKeyB64 || playerIdentityStrCopy == checkIdentityHash)
```

Comment:
I guess both base64 public key representation and hash should match, not only one of them?